framework/include/mesh/FaceInfo.h

```diff
@@ -87,14 +87,23 @@ class FaceInfo
   /// doubled in length.  The tip of this new vector is the neighbor centroid.
   /// This is important for FV dirichlet BCs.
   const Point & elemCentroid() const { return _elem_info->centroid(); }
-  const Point & neighborCentroid() const { return _neighbor_info->centroid(); }
+  const Point & neighborCentroid() const
+  {
+    mooseAssert(_neighbor_info,
+                "The neighbor is not defined on this faceInfo! A possible explanation is that the "
+                "face is a (physical/processor) boundary face.");
+    return _neighbor_info->centroid();
+  }
   ///@}
 
   ///@{
   /// Returns the elem and neighbor subdomain IDs. If no neighbor element exists, then
   /// an invalid ID is returned for the neighbor subdomain ID.
   SubdomainID elemSubdomainID() const { return _elem_info->subdomain_id(); }
-  SubdomainID neighborSubdomainID() const { return _neighbor_info->subdomain_id(); }
+  SubdomainID neighborSubdomainID() const
+  {
+    return _neighbor_info ? _neighbor_info->subdomain_id() : Moose::INVALID_BLOCK_ID;
```

Comment:
I know that you are in the draft phase, but I wanted to check over this PR to see if it might affect my porosity jump development ...

Any reason to not do the assertion treatment here that you are doing in the other methods you've modified?

Reply:
We should allow this because I think (I might be wrong here) in certain cases the subdomain IDs are used to identify if we are on a functor boundary or not.